pkg/server/server.go

```diff
@@ -709,6 +735,12 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 		utils.Bucketize(uint(resp.GetResolutionMetadata().DatastoreQueryCount), s.requestDurationByQueryHistogramBuckets),
 	).Observe(float64(time.Since(start).Milliseconds()))
 
+	requestDurationByQueryHistogram.WithLabelValues(
+		s.serviceName,
+		methodName,
+		utils.Bucketize(uint(resp.GetResolutionMetadata().DispatchCount), s.requestDurationByDispatchCountHistogramBuckets),
+	).Observe(float64(time.Since(start).Milliseconds()))
```

Comment:
i'm not very sure about how histograms work in prometheus so take this with a grain of salt, but shouldn't this be 

```go
requestDurationByQueryHistogram.WithLabelValues(
		s.serviceName,
		methodName,
                utils.Bucketize(uint(resp.GetResolutionMetadata().DatastoreQueryCount), s.requestDurationByQueryHistogramBuckets),
		utils.Bucketize(uint(resp.GetResolutionMetadata().DispatchCount), s.requestDurationByDispatchCountHistogramBuckets),
	).Observe(float64(time.Since(start).Milliseconds()))
```

?

since I assume that a) you need to specify a value for _all_ labels, and b) the label that you added `dispatch_count` comes last in the list? (is there a way to say more explicitly "this label has this value"??)

Reply:
Agree with @miparnisari - the way we add the dimension appears to be incorrect.